ONSdigital/response-operations-ui
# Strip whitespace from email address in EditContactDetailsForm

# Motivation and Context

There are two separate functions implemented for editing contact details at:
- `/edit-contact-details/<respondent_id>`
- `/<ru_ref>/edit-contact-details/<respondent_id>`
The former validates, but the other doesn't. Fixing validation in forms.py. Also changing email validation so that leading and trailing spaces are silently stripped, rather than throwing an error to the user: spaces in the email still raise an error in the edit dialogue.

# What has changed
Added a custom filter to the email field of `EditContactDetailsForm` in forms.py to str.strip() entries.  Added tests for this on respondent search and RU search.
Also bumped minor version of node package lodash to fix a medium-severity vulnerability.




# How to test?
- `make test` locally
- deploy gateway with new image to a sandbox cluster
- run acceptance tests on sandbox cluster 
- set port forwarding from localhost with ras-rm-cli
- edit contact details from respondent and RU search; ensure that spaces are silently stripped from both; ensure spaces in emails still raise an error.




# Links
- https://collaborate2.ons.gov.uk/jira/browse/RAS-40
- https://trello.com/c/kfHL2etU
